providers/profitbricks/src/main/java/org/jclouds/profitbricks/compute/extensions/ProfitBricksImageExtension.java

```diff
@@ -0,0 +1,139 @@
+ */
+package org.jclouds.profitbricks.compute.extensions;
+
+import static com.google.common.base.Preconditions.checkState;
+import static org.jclouds.profitbricks.config.ProfitBricksComputeProperties.TIMEOUT_SNAPSHOT_AVAILABLE;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.Callable;
+
+import javax.annotation.Resource;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.jclouds.Constants;
+import org.jclouds.compute.domain.CloneImageTemplate;
+import org.jclouds.compute.domain.Image;
+import org.jclouds.compute.domain.ImageTemplate;
+import org.jclouds.compute.domain.ImageTemplateBuilder;
+import org.jclouds.compute.extensions.ImageExtension;
+import org.jclouds.compute.reference.ComputeServiceConstants;
+import org.jclouds.logging.Logger;
+import org.jclouds.profitbricks.ProfitBricksApi;
+import org.jclouds.profitbricks.domain.Provisionable;
+import org.jclouds.profitbricks.domain.Server;
+import org.jclouds.profitbricks.domain.Snapshot;
+import org.jclouds.profitbricks.domain.Storage;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.UncheckedTimeoutException;
+import com.google.inject.Inject;
+
+@Singleton
+public class ProfitBricksImageExtension implements ImageExtension {
+
+   @Resource
+   @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+   protected Logger logger = Logger.NULL;
+
+   private final Predicate<Storage> matchBootDevice = new Predicate<Storage>() {
+      @Override
+      public boolean apply(Storage input) {
+         return input.bootDevice() == null ? false : input.bootDevice();
+      }
+   };
+
+   private final ProfitBricksApi api;
+   private final Predicate<String> snapshotAvailablePredicate;
+   private final Function<Provisionable, Image> imageTransformer;
+   private final ListeningExecutorService userExecutor;
+
+   @Inject
+   ProfitBricksImageExtension(ProfitBricksApi api,
+           @Named(TIMEOUT_SNAPSHOT_AVAILABLE) Predicate<String> snapshotAvailablePredicate,
+           Function<Provisionable, Image> imageTransformer,
+           @Named(Constants.PROPERTY_USER_THREADS) ListeningExecutorService userExecutor) {
+      this.api = api;
+      this.snapshotAvailablePredicate = snapshotAvailablePredicate;
+      this.imageTransformer = imageTransformer;
+      this.userExecutor = userExecutor;
+   }
+
+   @Override
+   public ImageTemplate buildImageTemplateFromNode(String name, String id) {
+      Server server = api.serverApi().getServer(id);
+
+      if (server == null)
+         throw new NoSuchElementException("Cannot find server with id: " + id);
+
+      List<Storage> storages = server.storages();
+      if (storages.isEmpty() || !Iterables.tryFind(storages, matchBootDevice).isPresent())
```

Comment:
`storages.isEmpty()` is redundant, you could remove it. Perhaps it reads better as: `if (!Iterables.any(storages, matchBootDevice))
`